### PR TITLE
Benchmark aggregate signatures and implementation improvements

### PR DESCRIPTION
**Pull request by jinmannwong**
Add benchmarks for aggregate signatures used in notarisations and improve implementation to improve time for constructing and verifying aggregate signatures.

### CODE REVIEW

File: libs/crypto/src/mcl_dkg.cpp [hunk 1]
```diff
@@ -400,14 +405,12 @@ PrivateKey SignatureAggregationCoefficient(PublicKey const &             notaris
 {
   PrivateKey coefficient;
 
+  std::string concatenated_keys;
+  concatenated_keys.reserve(48 + (cabinet_notarisation_keys.size() + 1) * 310);
```
Comment (n-hutton) [inline libs/crypto/src/mcl_dkg.cpp]: Seems unclear where 48 and 310 are from? Could they be constexpr somewhere? I am not sure I understand the hygenic reuse comment

**Review by n-hutton — CHANGES_REQUESTED** [review]
Minor change requested

File: libs/crypto/src/mcl_dkg.cpp (hunk 1 above)
Comment (jinmannwong) [reply libs/crypto/src/mcl_dkg.cpp]: The reserving of the 48 bytes was a figure David gave and the 310 is the size of each public key. I can make these const expressions. The hygenic reuse comment refers to the fact that the hash function (which is also used in the DKG) is being used here for a different purpose and to be completely safe we reserve some bytes at the front so there is no way two of the same strings could be hashed with the same hash function in the different contexts.

File: libs/crypto/benchmark/aggregate_signing.cpp [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    // Verify aggregate signature
+    auto aggregate_public_key = fetch::crypto::mcl::ComputeAggregatePublicKey(
+        aggregate_signature.second, notarisation_public_keys);
+    fetch::crypto::mcl::VerifySign(aggregate_public_key, msg, aggregate_signature.first, generator);
+  }
+}
+
+}  // namespace
+
+BENCHMARK(SignatureAggregationCoefficient)->RangeMultiplier(2)->Range(50, 500);
```
Comment (n-hutton) [inline libs/crypto/benchmark/aggregate_signing.cpp:232]: 👍

File: libs/crypto/src/mcl_dkg.cpp [hunk 3]
```diff
@@ -31,7 +31,8 @@ namespace fetch {
 namespace crypto {
 namespace mcl {
 
-std::atomic<bool> details::MCLInitialiser::was_initialised{false};
+std::atomic<bool>  details::MCLInitialiser::was_initialised{false};
+constexpr uint16_t public_key_byte_size = 310;
```
Comment (n-hutton) [inline libs/crypto/src/mcl_dkg.cpp]: This should be uppercase

**Review by n-hutton — CHANGES_REQUESTED** [review]
Minor change

**Review by n-hutton — APPROVED** [review]

**Pull request merged**